### PR TITLE
#59814:  descriptive back link appt details page

### PR DESCRIPTION
**Pull request by simiadebowale**
add descriptive back link to appointment details page.
screenshots:
upcoming:
Screenshot 2023-07-05 at 5 19 04 PM
localhost_3001_health-care_schedule-view-va-appointments_appointments_past_va_02aa456vvs(iPhone XR) (1)

past:
Screenshot 2023-07-05 at 5 22 23 PM
localhost_3001_health-care_schedule-view-va-appointments_appointments_past_va_02aa456vvs(iPhone XR)

pending: 
Screenshot 2023-07-05 at 5 23 00 PM
localhost_3001_health-care_schedule-view-va-appointments_appointments_past_va_02aa456vvs(iPhone XR) (2)

### CODE REVIEW

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/DetailsVideo.jsx [hunk 1]
```diff
@@ -6,7 +6,8 @@ import {
 } from '../../../services/appointment';
 import { VIDEO_TYPES } from '../../../utils/constants';
 import AppointmentDateTime from '../AppointmentDateTime';
-import Breadcrumbs from '../../../components/Breadcrumbs';
+import BackLink from '../../../components/BackLink';
+// import Breadcrumbs from '../../../components/Breadcrumbs';
```
Comment (vbahinwillit) [inline src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/DetailsVideo.jsx:10]: Remove unused code

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/DetailsVideo.jsx [hunk 2]
```diff
@@ -37,17 +38,18 @@ export default function DetailsVideo({ appointment, facilityData }) {
...
           }`}
         >
           Appointment detail
         </a>
-      </Breadcrumbs>
+      </Breadcrumbs> */}
+      <BackLink appointment={appointment} />
 
-      <h1>
+      <h1 style={{ marginTop: '2rem' }}>
```
Comment (vbahinwillit) [inline src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/DetailsVideo.jsx:52]: Use design system spacing unit '.vads-u-margin--0p25' which is the same as 2rem

File: src/applications/vaos/services/appointment/transformers.v2.js [hunk 3]
```diff
@@ -165,6 +165,21 @@ export function isPastAppointment(appt) {
...
+export function isUpcomingAppointment(appt, isRequest) {
+  const apptDateTime = moment(appt.start);
+  return (
+    !isRequest &&
+    !isPastAppointment(appt) &&
+    apptDateTime.isValid() &&
+    apptDateTime.isAfter(moment().startOf('day'))
+  );
+}
+
```
Comment (vbahinwillit) [inline src/applications/vaos/services/appointment/transformers.v2.js:182]: I would consider a different function name based on the description of what it does and the fact that there is another function in the code base with the same name.

../src/applications/vaos/services/appointment/index.js

File: src/applications/vaos/services/appointment/transformers.v2.js [hunk 4]
```diff
@@ -170,7 +170,7 @@ export function isPastAppointment(appt) {
  * @param {*} appt VAOS Service appointment object
  * @param {*} isRequest is appointment a request
  */
-export function isUpcomingAppointment(appt, isRequest) {
+export function isFutureAppointment(appt, isRequest) {
```
Comment (vbahinwillit) [inline src/applications/vaos/services/appointment/transformers.v2.js:173]: What's the difference between isUpcomingAppointment ../vaos/services/appointment/index.js:366? It looks like it does what you want (with additional checks).

Comment (simiadebowale) [reply src/applications/vaos/services/appointment/transformers.v2.js:173]: @vbahinwillit - the function you're referring to takes the already transformed appointment data as the arg. ie, `appt.vaos.appointmentType` on line 367. In this case, we need to determine if the appointment is upcoming before it is transformed (in the transformer)

Comment (vbahinwillit) [reply src/applications/vaos/services/appointment/transformers.v2.js:173]: So we now have 2 sources of truth for what's an upcoming appointment and it's computed differently. We have the selector, selectUpcomingAppointments and the new redux flag - appointments.confirmed[0].vaos.isUpcomingAppointment.

Comment (vbahinwillit) [reply src/applications/vaos/services/appointment/transformers.v2.js:173]: Couldn't the BackLink component use the selector?

Comment (simiadebowale) [reply src/applications/vaos/services/appointment/transformers.v2.js:173]: the `selectUpcomingAppointments` selector returns a list of upcoming appts, we don't need that here.

Comment (simiadebowale) [reply src/applications/vaos/services/appointment/transformers.v2.js:173]: @vbahinwillit - the selectUpcomingAppointments selector returns a list of upcoming appts, we don't need that here.

Comment (simiadebowale) [reply src/applications/vaos/services/appointment/transformers.v2.js:173]: @vbahinwillit - I can call `selectUpcomingAppointments ` in the backLink component and try to run a check for the appt ID for the appt details page. Having this in the transformer makes it reusable in the future.

**Review by vbahinwillit — APPROVED** [review]

**Pull request merged**